app/controllers/users_controller.rb

```diff
@@ -0,0 +1,26 @@
+class UsersController < ApplicationController
+  def edit
+    @user = current_user
+    @data = params[:user]
+    if(@data != nil)
+      @new_email = @data["new_email"]
+      if(!@new_email.to_s.empty?)
+        if(@new_email.match(/[a-zA-Z0-9._%]@(?:[a-zA-Z0-9]+\.)[a-zA-Z]{2,4}/))
+          @user.email = @new_email
+          @user.save
+          flash[:alert] = nil
```

Comment:
Tev nevajag iztirit flash. Vnk viņš turas vienā requestā ietvaros. Paskaties šeit par to 
https://guides.rubyonrails.org/action_controller_overview.html#the-flash

Tad šeit iekš condition tev sanaktu tā:

```
if @user.save
  flash[:notice] = 'all good'
  redirect_to root_path
else 
 flash.now[:alert] = 'nok'
 render :edit
end
```

Šeit ir tada viltiba: kā render viņš ielades tev template bet neparlodes mainigo no controllera, bet izmantos pašreizejo. Šeit vari palasit par to visu un kā skaisti jāizskatas kontrolerim.
https://guides.rubyonrails.org/getting_started.html#crudit-where-crudit-is-due